src/Microsoft.AspNetCore.Mvc.Core/ApiBehaviorOptions.cs

```diff
@@ -71,12 +71,12 @@ public ApiBehaviorOptions()
         public bool SuppressConsumesConstraintForFormFileParameters { get; set; }
 
         /// <summary>
-        /// Gets or sets a value that determines if controllers with <see cref="ApiControllerAttribute"/> use <see cref="ClientErrorFactory"/>
-        /// to transform certain certain client errors.
+        /// Gets or sets a value that determines if controllers with <see cref="ApiControllerAttribute"/>
+        /// transform certain certain client errors.
         /// <para>
-        /// When <c>false</c>, <see cref="ClientErrorFactory"/> is used to transform <see cref="IClientErrorActionResult"/> to the value
-        /// specified by the factory. In the default case, this converts <see cref="StatusCodeResult"/> instances to an <see cref="ObjectResult"/>
-        /// with <see cref="ProblemDetails"/>.
+        /// When <c>false</c>, a result filter is added to API controller actions that transforms <see cref="IClientErrorActionResult"/>.
+        /// By default, <see cref="ClientErrorMapping"/> is used to map <see cref="IClientErrorActionResult"/> to a
+        /// <see cref="ProblemDetails"/> instance (returned as the value for <see cref="ObjectResult"/>).
```

Comment:
This should explain how to replace the service if you want to use something other than problem details.